src/Features/Core/Diagnostics/BaseDiagnosticIncrementalAnalyzer.cs

```diff
@@ -166,6 +166,28 @@ protected BaseDiagnosticIncrementalAnalyzer(DiagnosticAnalyzerService owner, Wor
         public abstract Task<IEnumerable<DiagnosticData>> GetDiagnosticsForSpanAsync(Document document, TextSpan range, CancellationToken cancellationToken);
         #endregion
 
+        #region build error synchronization
+        /// <summary>
+        /// Callback from build listener.
+        /// 
+        /// Given diagnostics are errors host got from explicit build.
+        /// It is up to each analyzer how they will merge this information with live diagnostic info.
+        /// 
+        /// this API doesn't have cancellationToken since it can't be cancelled.
+        /// </summary>
+        public abstract Task SynchronizeWithBuildAsync(Project project, ImmutableArray<DiagnosticData> diagnostics);
```

Comment:
Its not clear from comments if these are project diagnostics (no location ones) or all project build diagnostics.